service/controller/chart/v1/resource/releasemigration/create.go

```diff
@@ -0,0 +1,80 @@
+package releasemigration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+// EnsureCreated ensures helm release is migrated from a v2 configmap to a v3 secret.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	hasConfigMap, err := r.getHelmV2ConfigMaps(ctx, key.ReleaseName(cr))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	hasSecret, err := r.getHelmV3Secrets(ctx, key.ReleaseName(cr), key.Namespace(cr))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// If Helm v2 release configmap had not been deleted and Helm v3 release secret is there,
+	// It means helm release migration is in progress.
+	if hasConfigMap && hasSecret {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q is under helmV3 migration", key.ReleaseName(cr)))
```

Comment:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q helmV3 migration in progress", key.ReleaseName(cr)))
```

I think this is clearer. As the next line is canceling reconciliation. WDYT?